### ISSUE TITLE
Konsole.Writer.MoveBufferArea throws System.PlatformNotSupportedException on Nix Platforms

### ISSUE THREAD

**Issue by Evelios** [goblinfactory/konsole#80]
I'm running into the exception `System.PlatformNotSupportedException` when running on linux. Through some sleuthing I discovered your stack overflow case [Is there any way to set the left and right margin of a console using Terminal sequences? the same way we can currently set Top and Bottom?](https://stackoverflow.com/questions/59555271/is-there-any-way-to-set-the-left-and-right-margin-of-a-console-using-terminal-se). From your issue it is not clear if you found an acceptable answer to your question. This also could be a duplicate of issue #30. I am compiling and running using .Net Core 3.1. If it is relevant am running the mate terminal and also need to support xterm for my users.

Relevant stack trace.
```
Unhandled exception. System.PlatformNotSupportedException: Operation is not supported on this platform.
   at System.ConsolePal.MoveBufferArea(Int32 sourceLeft, Int32 sourceTop, Int32 sourceWidth, Int32 sourceHeight, Int32 targetLeft, Int32 targetTop, Char sourceChar, ConsoleColor sourceForeColor, ConsoleColor sourceBackColor)
   at System.Console.MoveBufferArea(Int32 sourceLeft, Int32 sourceTop, Int32 sourceWidth, Int32 sourceHeight, Int32 targetLeft, Int32 targetTop, Char sourceChar, ConsoleColor sourceForeColor, ConsoleColor sourceBackColor)
   at Konsole.Writer.MoveBufferArea(Int32 sourceLeft, Int32 sourceTop, Int32 sourceWidth, Int32 sourceHeight, Int32 targetLeft, Int32 targetTop, Char sourceChar, ConsoleColor sourceForeColor, ConsoleColor sourceBackColor)
   at Konsole.ConcurrentWriter.MoveBufferArea(Int32 sourceLeft, Int32 sourceTop, Int32 sourceWidth, Int32 sourceHeight, Int32 targetLeft, Int32 targetTop, Char sourceChar, ConsoleColor sourceForeColor, ConsoleColor sourceBackColor)
   at Konsole.Window.ScrollDown()
   at Konsole.Window.WriteLine(String text)
   at Konsole.Window._write(String text)
   at Konsole.Window.Write(String text)
   at Konsole.ConcurrentWriter.Write(String text)
```

**Comment by goblinfactory** [comment]
Hi @Evelios 

I feel your pain, unfortunately there's some work that has to be done before this can be fixed. 
This is essentially a duplicate of #30 
https://github.com/goblinfactory/konsole/issues/30

i.e. the goal is to do important API refactors in version 7, new Input features in version 8, (keyboard handlers, maybe mouse support), and cross platform support in version 9.

Let me know your thoughts?

txs, 

Alan

**Comment by goblinfactory** [comment]
btw you can get around this issue by changing how your code writes with Konsole, if you want to keep using Konsole.
Set the window to Clip instead of scroll. So you can use Konsole to open a small window at location X, Y and have a progress bar in the window, print some status updates etc, use printAt to print at specific locations, benefit from being threadsafe etc.

**Comment by Evelios** [comment]
Thank you for your guidance, I'll close this issue as a duplicate to help you keep your plate clean

**Comment by goblinfactory** [comment]
thanks for taking the effort to report your concern. I hope to have a feature for this out quite soon, as a priority.
